[WIP][GSOC23] - E - Update CVE Audit documentation (#2471)
uyuni-project/uyuni-docs: modules/administration/pages/auditing.adoc

For more information about the patch status icons used on this page, see xref:reference:audit/audit-cve-audit.adoc[].

For each system, the [guimenu]``Actions`` column provides information about what you need to do to address vulnerabilities.
If applicable, a list of candidate channels or patches is also given.
You can also assign systems to a [guimenu]``System Set`` for further batch processing.

You can use the {productname} API to verify the patch status of your clients.
Use the ``audit.listSystemsByPatchStatus`` API method.
For more information about this method, see the {productname} API Guide.

== OVAL
The CVE Audit operation relies on two primary data sources: Channels and OVAL.
These two sources provide us with metadata for conducting CVE audits, each serving a distinct purpose.

> 1. **Channels:** Channels include the updated software packages, which include patches, and provide insights into the essential patches required to address vulnerabilities.
> 
> 2. **OVAL:** In contrast, OVAL data supplies information about the vulnerabilities themselves and the packages that render
> a system vulnerable to a CVE.

Reviewer: I'd prefer a definition list such as:

```suggestion
Channels::
Channels include the updated software packages, which include patches, and provide insights into the essential patches required to address vulnerabilities.

OVAL::
In contrast, OVAL data supplies information about the vulnerabilities themselves and the packages that render a system vulnerable to a CVE.
```

> While it is possible to conduct CVE audits using
> only channels data, synchronizing OVAL data enhances
> the accuracy of the results, particularly in cases involving zero-day vulnerabilities or partially patched
> vulnerabilities.

Reviewer: To better support the segmentation process for creating templates for localization (l10n), we prefer a 1-line per sentence style:

```suggestion
While it is possible to conduct CVE audits using only channels data, synchronizing OVAL data enhances the accuracy of the results, particularly in cases involving zero-day vulnerabilities or partially patched vulnerabilities.
```

> OVAL data is also much more lightweight than channels data .e.g. OVAL data for openSUSE Leap 15.4 is around ~50 MB.

Reviewer: ```suggestion
OVAL data is also much more lightweight than channels data.
For example, OVAL data for {opensuse} Leap 15.4 is around 50{nbsp}MB.
```

Having synced OVAL data only, you can already perform CVE audits and check if your systems are vulnerable or not to a CVE, but you can't apply patches since they come from channels.

> By default, OVAL data, is updated at 23:00 every day. We recommend that before you begin a CVE audit you refresh the data to ensure you have the latest vulnerabilities metadata.

Reviewer: ```suggestion
By default, OVAL data is updated at 23:00 every day.
We recommend that before you begin a CVE audit you refresh the data to ensure you have the latest vulnerabilities metadata.
```

.Procedure: Updating OVAL Data
. In the {productname} {webui}, navigate to menu:Admin[Task Schedules] and select the ``oval-data-sync-default`` schedule.
. Click btn:[oval-data-sync-bunch].
. Click btn:[Single Run Schedule] to schedule the task.

> Allow the task to complete before continuing with the CVE audit.

Reviewer: ```suggestion

Allow the task to complete before continuing with the CVE audit.
```

=== Collecting CPE

To be able to accurately identify what vulnerabilities apply to a certain client, we need to identify the operating system product that client uses. To do that, we collect the CPE (Common Platform Enumeration) of the client as a salt grain, then we save it to the database.

> The cpe of newly registered clients will be automatically collected and saved to the database. However, for existing clients, it is necessary to execute the ``Update Packages List`` action at least once to

Reviewer: I'm not sure about the sentence end.  Here is my proposal
```suggestion
The CPE of newly registered clients will be automatically collected and saved to the database.
However, for existing clients, it is necessary to execute the ``Update Packages List`` action at least once.
```
